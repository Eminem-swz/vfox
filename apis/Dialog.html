<!DOCTYPE html>
<html lang="zh-CN">
  <head>
    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width,initial-scale=1">
    <title>Dialog 模态对话框 | Vfox for Vue3</title>
    <meta name="description" content="移动端 Vue 组件库">
    <link rel="stylesheet" href="/vfox/assets/style.cd17bc9c.css">
    <link rel="modulepreload" href="/vfox/assets/Home.ef41725c.js">
    <link rel="modulepreload" href="/vfox/assets/app.8c4fff02.js">
    <link rel="modulepreload" href="/vfox/assets/apis_Dialog.md.1016d91e.lean.js">
    
    <meta name="twitter:title" content="Dialog 模态对话框 | Vfox for Vue3">
  <meta property="og:title" content="Dialog 模态对话框 | Vfox for Vue3">
  </head>
  <body>
    <div id="app"><!--[--><div class="theme"><header class="nav-bar" data-v-cbbf79d8><div class="sidebar-button" data-v-cbbf79d8><svg class="icon" xmlns="http://www.w3.org/2000/svg" aria-hidden="true" role="img" viewBox="0 0 448 512"><path fill="currentColor" d="M436 124H12c-6.627 0-12-5.373-12-12V80c0-6.627 5.373-12 12-12h424c6.627 0 12 5.373 12 12v32c0 6.627-5.373 12-12 12zm0 160H12c-6.627 0-12-5.373-12-12v-32c0-6.627 5.373-12 12-12h424c6.627 0 12 5.373 12 12v32c0 6.627-5.373 12-12 12zm0 160H12c-6.627 0-12-5.373-12-12v-32c0-6.627 5.373-12 12-12h424c6.627 0 12 5.373 12 12v32c0 6.627-5.373 12-12 12z" class></path></svg></div><a class="nav-bar-title" href="/vfox/" aria-label="Vfox for Vue3, back to home" data-v-cbbf79d8 data-v-3902cc82><!----> Vfox for Vue3</a><div class="flex-grow" data-v-cbbf79d8></div><div class="nav" data-v-cbbf79d8><nav class="nav-links" data-v-cbbf79d8 data-v-6c4348e2><!--[--><div class="item" data-v-6c4348e2><div class="nav-link" data-v-6c4348e2 data-v-3efd0139><a class="item" href="/vfox/" data-v-3efd0139>指南 <!----></a></div></div><div class="item" data-v-6c4348e2><div class="nav-link" data-v-6c4348e2 data-v-3efd0139><a class="item" href="/vfox/design/" data-v-3efd0139>设计 <!----></a></div></div><div class="item" data-v-6c4348e2><div class="nav-link" data-v-6c4348e2 data-v-3efd0139><a class="item" href="/vfox/components/Button" data-v-3efd0139>组件 <!----></a></div></div><div class="item" data-v-6c4348e2><div class="nav-link" data-v-6c4348e2 data-v-3efd0139><a class="item active" href="/vfox/apis/Toast" data-v-3efd0139>API <!----></a></div></div><div class="item" data-v-6c4348e2><div class="nav-dropdown-link" data-v-6c4348e2 data-v-a5ec8504><button class="button" data-v-a5ec8504><span class="button-text" data-v-a5ec8504>选择版本</span><span class="right button-arrow" data-v-a5ec8504></span></button><ul class="dialog" data-v-a5ec8504><!--[--><li class="dialog-item" data-v-a5ec8504><div class="nav-dropdown-link-item" data-v-a5ec8504 data-v-5a696df7><a class="item isExternal" href="https://cdn.fox2.cn/1.x/docs/" target="_blank" rel="noopener noreferrer" data-v-5a696df7><span class="arrow" data-v-5a696df7></span><span class="text" data-v-5a696df7>1.x</span><span class="icon" data-v-5a696df7><svg class="icon outbound" xmlns="http://www.w3.org/2000/svg" aria-hidden="true" x="0px" y="0px" viewbox="0 0 100 100" width="15" height="15" data-v-5a696df7><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path><polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg></span></a></div></li><li class="dialog-item" data-v-a5ec8504><div class="nav-dropdown-link-item" data-v-a5ec8504 data-v-5a696df7><a class="item active" href="/vfox/" data-v-5a696df7><span class="arrow" data-v-5a696df7></span><span class="text" data-v-5a696df7>2.x</span><span class="icon" data-v-5a696df7><!----></span></a></div></li><!--]--></ul></div></div><!--]--><!----><!----></nav></div><!--[--><!--]--></header><aside class="sidebar" data-v-722cd14b><nav class="nav-links nav" data-v-722cd14b data-v-6c4348e2><!--[--><div class="item" data-v-6c4348e2><div class="nav-link" data-v-6c4348e2 data-v-3efd0139><a class="item" href="/vfox/" data-v-3efd0139>指南 <!----></a></div></div><div class="item" data-v-6c4348e2><div class="nav-link" data-v-6c4348e2 data-v-3efd0139><a class="item" href="/vfox/design/" data-v-3efd0139>设计 <!----></a></div></div><div class="item" data-v-6c4348e2><div class="nav-link" data-v-6c4348e2 data-v-3efd0139><a class="item" href="/vfox/components/Button" data-v-3efd0139>组件 <!----></a></div></div><div class="item" data-v-6c4348e2><div class="nav-link" data-v-6c4348e2 data-v-3efd0139><a class="item active" href="/vfox/apis/Toast" data-v-3efd0139>API <!----></a></div></div><div class="item" data-v-6c4348e2><div class="nav-dropdown-link" data-v-6c4348e2 data-v-a5ec8504><button class="button" data-v-a5ec8504><span class="button-text" data-v-a5ec8504>选择版本</span><span class="right button-arrow" data-v-a5ec8504></span></button><ul class="dialog" data-v-a5ec8504><!--[--><li class="dialog-item" data-v-a5ec8504><div class="nav-dropdown-link-item" data-v-a5ec8504 data-v-5a696df7><a class="item isExternal" href="https://cdn.fox2.cn/1.x/docs/" target="_blank" rel="noopener noreferrer" data-v-5a696df7><span class="arrow" data-v-5a696df7></span><span class="text" data-v-5a696df7>1.x</span><span class="icon" data-v-5a696df7><svg class="icon outbound" xmlns="http://www.w3.org/2000/svg" aria-hidden="true" x="0px" y="0px" viewbox="0 0 100 100" width="15" height="15" data-v-5a696df7><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path><polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg></span></a></div></li><li class="dialog-item" data-v-a5ec8504><div class="nav-dropdown-link-item" data-v-a5ec8504 data-v-5a696df7><a class="item active" href="/vfox/" data-v-5a696df7><span class="arrow" data-v-5a696df7></span><span class="text" data-v-5a696df7>2.x</span><span class="icon" data-v-5a696df7><!----></span></a></div></li><!--]--></ul></div></div><!--]--><!----><!----></nav><!--[--><!--]--><ul class="sidebar-links" data-v-722cd14b><!--[--><li class="sidebar-link"><p class="sidebar-link-item">通知类弹窗</p><ul class="sidebar-links"><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/apis/Toast">Toast</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/apis/Notify">Notify</a><!----></li></ul></li><li class="sidebar-link"><p class="sidebar-link-item">预览类弹窗</p><ul class="sidebar-links"><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/apis/Preview">Preview</a><!----></li></ul></li><li class="sidebar-link"><p class="sidebar-link-item">模态框类弹窗</p><ul class="sidebar-links"><li class="sidebar-link"><a class="sidebar-link-item active" href="/vfox/apis/Dialog">Dialog</a><ul class="sidebar-links"><li class="sidebar-link"><a class="sidebar-link-item" href="#showdialog-object">showDialog(object)</a><ul class="sidebar-links"><li class="sidebar-link"><a class="sidebar-link-item" href="#object">object</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="#usage">Usage</a><!----></li></ul></li></ul></li></ul></li><li class="sidebar-link"><p class="sidebar-link-item">抽屉类弹窗</p><ul class="sidebar-links"><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/apis/ActionSheet">ActionSheet</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/apis/Picker">Picker</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/apis/DatePicker">DatePicker</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/apis/Cascader">Cascader</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/apis/Calendar">Calendar</a><!----></li></ul></li><li class="sidebar-link"><p class="sidebar-link-item">气泡类弹窗</p><ul class="sidebar-links"><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/apis/Popover">Popover</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/apis/PopDialog">PopDialog</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/apis/PopMenu">PopMenu</a><!----></li></ul></li><!--]--></ul><!--[--><!--]--></aside><div class="sidebar-mask"></div><main class="page" data-v-0fc60803><div class="container" data-v-0fc60803><!--[--><!--[--><!----><!--]--><!--[--><div class="demo"><iframe style="display:none;" src="https://godxiaoji.github.io/vfox/demo/#/" frameborder="0"></iframe></div><!--]--><!--]--><div style="position:relative;" class="content" data-v-0fc60803><div><h1 id="dialog-模态对话框" tabindex="-1">Dialog 模态对话框 <a class="header-anchor" href="#dialog-模态对话框" aria-hidden="true">#</a></h1><h2 id="showdialog-object" tabindex="-1">showDialog(object) <a class="header-anchor" href="#showdialog-object" aria-hidden="true">#</a></h2><p>显示模态对话框。</p><h3 id="object" tabindex="-1">object <a class="header-anchor" href="#object" aria-hidden="true">#</a></h3><table><thead><tr><th>属性</th><th>类型</th><th>默认值</th><th>必填</th><th>说明</th></tr></thead><tbody><tr><td>title</td><td>string</td><td></td><td>否</td><td>提示的标题</td></tr><tr><td>content</td><td>string</td><td></td><td>是</td><td>提示的内容</td></tr><tr><td>maskClosable</td><td>boolean</td><td>false</td><td>否</td><td>点击蒙层是否触发关闭操作</td></tr><tr><td>showCancel</td><td>boolean</td><td>true</td><td>否</td><td>是否显示取消按钮</td></tr><tr><td>confirmText</td><td>string</td><td>&#39;确定&#39;</td><td>否</td><td>确认按钮的文字</td></tr><tr><td>cancelText</td><td>string</td><td>&#39;取消&#39;</td><td>否</td><td>取消按钮的文字</td></tr><tr><td>success</td><td>function</td><td></td><td>否</td><td>接口调用成功（在用户做出选择后，如取消，确定）的回调函数</td></tr><tr><td>fail</td><td>function</td><td></td><td>否</td><td>接口调用失败（如传入错误的参数）的回调函数（不传入 fail 遇错误直接抛出）</td></tr><tr><td>complete</td><td>function</td><td></td><td>否</td><td>弹窗关闭或调用失败的回调函数</td></tr></tbody></table><h4 id="object-success-回调参数" tabindex="-1">object.success 回调参数 <a class="header-anchor" href="#object-success-回调参数" aria-hidden="true">#</a></h4><table><thead><tr><th>属性</th><th>类型</th><th>说明</th></tr></thead><tbody><tr><td>confirm?</td><td>boolean</td><td>为 true 时，表示用户点击了确定按钮</td></tr><tr><td>cancel?</td><td>boolean</td><td>为 true 时，表示用户取消了</td></tr></tbody></table><h3 id="usage" tabindex="-1">Usage <a class="header-anchor" href="#usage" aria-hidden="true">#</a></h3><p>具体调用方式可以参考<a href="./../guide/import.html#api-调用">API 调用</a>。</p><div class="language-JavaScript"><pre><code><span class="token function">showDialog</span><span class="token punctuation">(</span><span class="token punctuation">{</span>
  <span class="token literal-property property">title</span><span class="token operator">:</span> <span class="token string">&#39;提示&#39;</span><span class="token punctuation">,</span>
  <span class="token literal-property property">content</span><span class="token operator">:</span> <span class="token string">&#39;这是一个模态弹窗&#39;</span><span class="token punctuation">,</span>
  <span class="token function">success</span> <span class="token punctuation">(</span><span class="token parameter">res</span><span class="token punctuation">)</span> <span class="token punctuation">{</span>
    <span class="token keyword">if</span> <span class="token punctuation">(</span>res<span class="token punctuation">.</span>confirm<span class="token punctuation">)</span> <span class="token punctuation">{</span>
      console<span class="token punctuation">.</span><span class="token function">log</span><span class="token punctuation">(</span><span class="token string">&#39;用户点击确定&#39;</span><span class="token punctuation">)</span>
    <span class="token punctuation">}</span> <span class="token keyword">else</span> <span class="token keyword">if</span> <span class="token punctuation">(</span>res<span class="token punctuation">.</span>cancel<span class="token punctuation">)</span> <span class="token punctuation">{</span>
      console<span class="token punctuation">.</span><span class="token function">log</span><span class="token punctuation">(</span><span class="token string">&#39;用户点击取消&#39;</span><span class="token punctuation">)</span>
    <span class="token punctuation">}</span>
  <span class="token punctuation">}</span>
<span class="token punctuation">}</span><span class="token punctuation">)</span>
</code></pre></div></div></div><footer class="page-footer" data-v-0fc60803 data-v-e4f5d8c8><div class="edit" data-v-e4f5d8c8><div class="edit-link" data-v-e4f5d8c8 data-v-25f5f3d4><!----></div></div><div class="updated" data-v-e4f5d8c8><!----></div></footer><div class="next-and-prev-link" data-v-0fc60803 data-v-162a74c0><div class="container" data-v-162a74c0><div class="prev" data-v-162a74c0><a class="link" href="/vfox/apis/Preview" data-v-162a74c0><svg xmlns="http://www.w3.org/2000/svg" viewbox="0 0 24 24" class="icon icon-prev" data-v-162a74c0><path d="M19,11H7.4l5.3-5.3c0.4-0.4,0.4-1,0-1.4s-1-0.4-1.4,0l-7,7c-0.1,0.1-0.2,0.2-0.2,0.3c-0.1,0.2-0.1,0.5,0,0.8c0.1,0.1,0.1,0.2,0.2,0.3l7,7c0.2,0.2,0.5,0.3,0.7,0.3s0.5-0.1,0.7-0.3c0.4-0.4,0.4-1,0-1.4L7.4,13H19c0.6,0,1-0.4,1-1S19.6,11,19,11z"></path></svg><span class="text" data-v-162a74c0>Preview</span></a></div><div class="next" data-v-162a74c0><a class="link" href="/vfox/apis/ActionSheet" data-v-162a74c0><span class="text" data-v-162a74c0>ActionSheet</span><svg xmlns="http://www.w3.org/2000/svg" viewbox="0 0 24 24" class="icon icon-next" data-v-162a74c0><path d="M19.9,12.4c0.1-0.2,0.1-0.5,0-0.8c-0.1-0.1-0.1-0.2-0.2-0.3l-7-7c-0.4-0.4-1-0.4-1.4,0s-0.4,1,0,1.4l5.3,5.3H5c-0.6,0-1,0.4-1,1s0.4,1,1,1h11.6l-5.3,5.3c-0.4,0.4-0.4,1,0,1.4c0.2,0.2,0.5,0.3,0.7,0.3s0.5-0.1,0.7-0.3l7-7C19.8,12.6,19.9,12.5,19.9,12.4z"></path></svg></a></div></div></div><!--[--><!--]--></div></main></div><!----><!--]--></div>
    <script>__VP_HASH_MAP__ = JSON.parse("{\"apis_actionsheet.md\":\"21036791\",\"apis_calendar.md\":\"db5bc3c0\",\"apis_cascader.md\":\"bc7e103b\",\"apis_datepicker.md\":\"aced03f1\",\"apis_dialog.md\":\"1016d91e\",\"apis_notify.md\":\"420dc611\",\"apis_picker.md\":\"c7589b37\",\"apis_popdialog.md\":\"56b2612b\",\"apis_popmenu.md\":\"9c9b75d3\",\"apis_popover.md\":\"185631df\",\"apis_preview.md\":\"2c9c072b\",\"apis_toast.md\":\"48cdfdcb\",\"apis_index.md\":\"07408e35\",\"components_actionsheet.md\":\"d8f917c1\",\"components_activityindicator.md\":\"091fde5f\",\"components_avatar.md\":\"89cbcc10\",\"components_backtop.md\":\"c6cdebce\",\"components_badge.md\":\"ef887f05\",\"components_button.md\":\"ade99656\",\"components_calendar.md\":\"5e0a4e64\",\"components_cascader.md\":\"995d328e\",\"components_cell.md\":\"98482ca5\",\"components_checkbox.md\":\"fbcc7c0a\",\"components_circleprogress.md\":\"3aefedf2\",\"components_collapse.md\":\"0d31a4e6\",\"components_copy.md\":\"17e411f2\",\"components_countdown.md\":\"9a227150\",\"components_countup.md\":\"c9526fce\",\"components_datepicker.md\":\"e2daaebc\",\"components_dialog.md\":\"8ebb4361\",\"components_divider.md\":\"4ad57490\",\"components_drawer.md\":\"78ae5254\",\"components_dropdown.md\":\"febfb980\",\"components_empty.md\":\"c0547323\",\"components_fixed.md\":\"08e82b39\",\"components_flatlist.md\":\"2e3e2be0\",\"components_form.md\":\"455d5d9e\",\"components_group.md\":\"5d9e9051\",\"components_icon.md\":\"e8f60612\",\"components_image.md\":\"6b91732e\",\"components_imagepreview.md\":\"19257825\",\"components_imageuploader.md\":\"4c3e9028\",\"components_indexview.md\":\"cc90d2bc\",\"components_input.md\":\"ffcbd30c\",\"components_layout.md\":\"f47a9b34\",\"components_loadmore.md\":\"9d51cd3d\",\"components_modal.md\":\"c9f7d1bf\",\"components_navbar.md\":\"a053f9dc\",\"components_noticebar.md\":\"cf249cc9\",\"components_notify.md\":\"9fe7c840\",\"components_numberkeyboard.md\":\"1e7eeb85\",\"components_order.md\":\"e0acc42a\",\"components_pagination.md\":\"8a25ab75\",\"components_picker.md\":\"50a226f2\",\"components_popdialog.md\":\"781d04d9\",\"components_popmenu.md\":\"f5a98860\",\"components_popover.md\":\"58854355\",\"components_price.md\":\"b31abd1b\",\"components_progress.md\":\"d134ebce\",\"components_radio.md\":\"288f7517\",\"components_range.md\":\"8793cb97\",\"components_rate.md\":\"faff522a\",\"components_result.md\":\"019536b5\",\"components_scrolltab.md\":\"0d91b90b\",\"components_scrollview.md\":\"03b6b136\",\"components_searchbar.md\":\"6cd36bb4\",\"components_sidetab.md\":\"d3201b56\",\"components_skeleton.md\":\"8159aa3e\",\"components_slider.md\":\"87257653\",\"components_stepper.md\":\"4eb881a9\",\"components_steps.md\":\"1208a6d7\",\"components_sticky.md\":\"e6b83e11\",\"components_stopwatch.md\":\"27215d5e\",\"components_swipecell.md\":\"52e3c823\",\"components_swiper.md\":\"83997734\",\"components_switch.md\":\"e3446337\",\"components_tab.md\":\"1b96f195\",\"components_tabbar.md\":\"4976b686\",\"components_tabview.md\":\"b1d46eb0\",\"components_tag.md\":\"dd6d791a\",\"components_timeago.md\":\"50869e34\",\"components_timeline.md\":\"0655d409\",\"components_toast.md\":\"8131e42c\",\"components_index.md\":\"1a742709\",\"design_color.md\":\"0494c5f2\",\"design_index.md\":\"f72698ef\",\"guide_dark.md\":\"10f25831\",\"guide_import.md\":\"607a8c2b\",\"guide_locale.md\":\"3166a29e\",\"guide_theme.md\":\"f3bc5373\",\"index.md\":\"7d77b50b\"}")</script>
    <script type="module" async src="/vfox/assets/app.8c4fff02.js"></script>
    
  </body>
</html>