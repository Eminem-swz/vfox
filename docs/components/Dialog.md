# Dialog 模态对话框

## Import

```
import { Dialog } from 'vfox'
```

具体的引入方式可以参考[引入组件](../guide/import.md)。

## Props

| 属性            | 类型    | 默认值 | 必填 | 说明                     |
| --------------- | ------- | ------ | ---- | ------------------------ |
| v-model:visible | boolean | false  | 否   | 是否显示                 |
| title           | string  | '提示' | 否   | 提示的标题               |
| content         | string  |        | 否   | 提示的内容，优先于 slot  |
| mask-closable   | boolean | false  | 否   | 点击蒙层是否触发关闭操作 |
| show-cancel     | boolean | true   | 否   | 是否显示取消按钮         |
| confirm-text    | string  | '确定' | 否   | 确认按钮的文字           |
| cancel-text     | string  | '取消' | 否   | 取消按钮的文字           |

## Events

| 事件                 | 描述               | 回调函数参数                                               | 函数 TypeScript           |
| -------------------- | ------------------ | ---------------------------------------------------------- | ------------------------- |
| cancel               | 取消按钮点击时触发 | { source: string }                                         | PopupOnCancel             |
| confirm              | 确认按钮点击时触发 | { }                                                        |                           |
| visible-state-change | 展示隐藏时触发     | { state: [VisibleState](./Dialog.md#visiblestate-值说明) } | PopupOnVisibleStateChange |

### VisibleState 值说明

| 值     | 说明                 | 备注                                              |
| ------ | -------------------- | ------------------------------------------------- |
| show   | 展示时触发           |                                                   |
| shown  | 展示且动画结束后触发 |                                                   |
| hide   | 隐藏时触发           | 可能携带其他参数 cancel, maskClick, closeClick 等 |
| hidden | 隐藏且动画结束后触发 | 可能携带其他参数 cancel, maskClick, closeClick 等 |

## Slots

### 默认（#default）

```
<fx-dialog>
  提示内容
  ...
</fx-dialog>
```
