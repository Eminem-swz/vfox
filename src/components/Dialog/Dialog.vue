<template>
  <modal
    :class="[prefix + '-dialog']"
    :visible.sync="visible2"
    :show-close="false"
    :mask-closable="maskClosable"
    @show="onShow"
    @shown="onShown"
    @hide="onHide"
    @hidden="onHidden"
    @cancel="onCancel"
    ref="popup"
  >
    <div :class="[prefix + '-dialog_header']" v-if="title != null">
      {{ title }}
    </div>
    <div :class="[prefix + '-dialog_content']">
      <div :class="[prefix + '-dialog_content-text']" v-if="content != null">
        {{ content }}
      </div>
      <slot v-else></slot>
    </div>
    <div :class="[prefix + '-dialog_footer', prefix + '-horizontal-hairline']">
      <fx-button-group :class="[prefix + '-dialog_footer-inner']" pattern="borderless">
        <fx-button v-if="showCancel" :class="[prefix + '-dialog_button']" type="default" @click="onCancelClick">
          {{ cancelText }}
        </fx-button>
        <fx-button :class="[prefix + '-dialog_button']" type="primary" @click="onConfirmClick">
          {{ confirmText }}
        </fx-button>
      </fx-button-group>
    </div>
  </modal>
</template>

<script>
import FxButton from '../Button/Button.vue'
import FxButtonGroup from '../Button/Group.vue'
import Modal from '../Modal/Modal.vue'
import { SDKKey } from '../../config'
import popupExtendMixin from '../util/popup-extend-mixin'

export default {
  name: SDKKey + '-dialog',
  components: { FxButton, FxButtonGroup, Modal },
  mixins: [popupExtendMixin],
  props: {
    title: {
      type: String,
      default: null
    },
    cancelText: {
      type: String,
      default: '取消'
    },
    confirmText: {
      type: String,
      default: '确定'
    },
    maskClosable: {
      type: Boolean,
      default: false
    },
    showCancel: {
      type: Boolean,
      default: true
    },
    content: {
      type: String,
      default: null
    }
  },
  data() {
    return { prefix: SDKKey }
  },
  watch: {},
  created() {},
  beforeDestroy() {},
  methods: {
    onConfirmClick() {
      this.$emit('confirm', {})
      this.afterConfirm({})

      this.visible2 = false
    }
  }
}
</script>

<style lang="scss">
@import '../component.module.scss';

.#{$prefix}-dialog {
  &_header {
    flex: 1;
    font-size: 17px;
    line-height: 24px;
    height: 24px;
    font-weight: 500;
    color: $title-color;
    margin: 32px 0 0;
    display: -webkit-box;
    -webkit-line-clamp: 1;
    -webkit-box-orient: vertical;
    overflow: hidden;
    text-align: center;
    white-space: pre-line;

    + .#{$prefix}-dialog_content {
      margin-top: 0;
    }
  }

  &_content {
    margin-top: 16px;
    padding: 16px 24px;
    font-size: 17px;
    line-height: 24px;
    color: $font-color;

    &-text {
      min-height: 24px;
      max-height: 300px;
      overflow-x: hidden;
      overflow-y: auto;
      word-break: break-all;
      word-wrap: break-word;
      text-align: center;
      white-space: pre-line;
    }
  }

  &_footer {
    display: flex;
    flex-direction: column;
    margin-top: 16px;

    &::before {
      content: '';
    }

    &-inner {
      display: flex;
      justify-content: space-between;
      position: relative;

      > .#{$prefix}-button + .#{$prefix}-button::before {
        height: 100%;
        background-color: $border-color;
      }
    }
  }
}
</style>
