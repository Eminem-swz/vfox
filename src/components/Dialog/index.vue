<template>
  <modal
    :visible.sync="visible"
    @click="onClose"
    :show-close="false"
    :z-index="zIndex"
    :mask-closable="maskClosable"
    ref="modal"
  >
    <template #header>
      {{ title }}
    </template>
    <div :class="[prefix + '-dialog_content']">
      <template v-if="content">
        {{ content }}
      </template>
      <slot v-else></slot>
    </div>
    <template #footer>
      <fx-button
        v-show="showCancel"
        :class="[prefix + '-modal_button']"
        type="default"
        @click="onCancelClick"
      >
        {{ cancelText }}
      </fx-button>
      <fx-button
        :class="[prefix + '-modal_button']"
        type="primary"
        @click="onConfirmClick"
      >
        {{ confirmText }}
      </fx-button>
    </template>
  </modal>
</template>

<script>
import FxButton from '../Button'
import Modal from '../Modal'
import { CustomEvent } from '../../helpers/events'
import { SDKKey } from '../../config'

export default {
  name: SDKKey + '-dialog',
  components: { FxButton, Modal },
  props: {
    visible: {
      type: Boolean,
      default: false
    },
    title: {
      type: String,
      default: '提示'
    },
    cancelText: {
      type: String,
      default: '取消'
    },
    confirmText: {
      type: String,
      default: '确定'
    },
    maskClosable: {
      type: Boolean,
      default: false
    },
    showCancel: {
      type: Boolean,
      default: true
    },
    zIndex: {
      type: Number,
      default: 2000
    },
    content: {
      type: String,
      default: ''
    }
  },
  data() {
    return { prefix: SDKKey }
  },
  created() {},
  beforeDestroy() {},
  methods: {
    onCancelClick(e) {
      const type = 'cancel'

      this.$emit(
        type,
        new CustomEvent(
          {
            type,
            currentTarget: this.$el
          },
          {}
        )
      )

      this.$refs.close.close(e)
    },
    onConfirmClick(e) {
      const type = 'confirm'

      this.$emit(
        type,
        new CustomEvent(
          {
            type,
            currentTarget: this.$el
          },
          {}
        )
      )

      this.$refs.close.close(e)
    },
    onClose() {
      this.$emit('update:visible', false)

      const type = 'close'

      this.$emit(
        type,
        new CustomEvent(
          {
            type,
            currentTarget: this.$el
          },
          {}
        )
      )
    }
  }
}
</script>

<style lang="scss">
@import '../component.module.scss';

.#{$prefix}-dialog {
  &_content {
    padding: 16px 24px;
    max-height: 300px;
    overflow-x: hidden;
    overflow-y: auto;
    word-break: break-all;
    word-wrap: break-word;
    text-align: center;
  }
}
</style>
