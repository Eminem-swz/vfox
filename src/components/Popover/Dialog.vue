<template>
  <div
    :class="[
      prefix + '-popover',
      prefix + '-pop-dialog',
      prefix + '-popup',
      { visible: visible2, 'no--mask': !showMask }
    ]"
    :style="{ zIndex }"
    v-show="isShow"
  >
    <div :class="[prefix + '-mask']" @click="onMaskClick"></div>
    <div :class="[prefix + '-popover_inner']" ref="inner" :style="innerStyles">
      <i :class="[prefix + '-popover_arrow']" :style="arrowStyles"></i>
      <div :class="[prefix + '-popover_content']">
        <div :class="[prefix + '-pop-dialog_body']">
          <div :class="[prefix + '-popover_text']">{{ content }}</div>
        </div>
        <div :class="[prefix + '-pop-dialog_footer', prefix + '-horizontal-hairline']">
          <fx-button-group :class="[prefix + '-pop-dialog_footer-inner']" pattern="borderless" size="middle">
            <fx-button v-if="showCancel" type="default" @click="onCancelClick">
              {{ cancelText }}
            </fx-button>
            <fx-button type="primary" @click="onConfirmClick">
              {{ confirmText }}
            </fx-button>
          </fx-button-group>
        </div>
      </div>
    </div>
  </div>
</template>

<script>
import Popover from './Popover.vue'
import FxButton from '../Button/Button.vue'
import FxButtonGroup from '../Button/Group.vue'
import { SDKKey } from '../../config'

export default {
  name: SDKKey + '-pop-dialog',
  extends: Popover,
  components: { FxButton, FxButtonGroup },
  props: {
    content: {
      type: String,
      default: '',
      required: true
    },
    showCancel: {
      type: Boolean,
      default: true
    },
    cancelText: {
      type: String,
      default: '取消'
    },
    confirmText: {
      type: String,
      default: '确定'
    }
  },
  computed: {},
  methods: {
    onConfirmClick() {
      this.$emit('confirm', {})
      this.afterConfirm({})

      this.hide()
    },
    afterConfirm() {}
  }
}
</script>

<style lang="scss">
@import '../component.module.scss';

.#{$prefix}-pop-dialog {
  &_footer {
    display: flex;
    flex-direction: column;

    &::before {
      content: '';
    }

    &-inner {
      display: flex;
      justify-content: space-between;
      position: relative;

      > .#{$prefix}-button + .#{$prefix}-button::before {
        height: 100%;
        background-color: $border-color;
      }
    }
  }
}
</style>
