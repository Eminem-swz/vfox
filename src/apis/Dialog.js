import Vue from 'vue'
import Dialog from '../components/Dialog'
import { createPopup } from '../helpers/popup'
import { htmlEscape, isObject } from '../helpers/util'
import { getCallbackFns } from './callback'
import { parseParamsByRules } from './rules'
import { removeEl } from '../helpers/dom'

export function showDialog(object) {
  if (!isObject(object)) {
    object = {}
  }

  const { success, fail, complete } = getCallbackFns(object)

  try {
    const propsData = parseParamsByRules(object, 'showDialog')
    propsData.content = htmlEscape(propsData.content)

    const Comp = Vue.extend({
      extends: Dialog,
      methods: {
        onCancelClick(e) {
          const res = {
            confirm: false,
            cancel: true
          }

          success(res)
          complete()

          this.close(e)
        },
        onConfirmClick(e) {
          const res = {
            confirm: true,
            cancel: false
          }

          success(res)
          complete()

          this.close(e)
        },
        close() {
          this.visible = false
          const $el = this.$el

          setTimeout(() => {
            this.$destroy()

            removeEl($el)
          }, 200)
        }
      }
    })

    const { $wrapper, zIndex } = createPopup()

    const app = new Comp({
      propsData: Object.assign(propsData, {
        zIndex,
        visible: true
      })
    }).$mount($wrapper)

    return app
  } catch (e) {
    fail(e)
    complete()
  }
}
