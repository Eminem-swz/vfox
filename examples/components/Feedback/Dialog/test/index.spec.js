/* global describe,test,expect */

import { mount } from '@vue/test-utils'
import Dialog from '@/Dialog'

describe('Dialog', () => {
  test('should render default correctly', () => {
    const wrapper = mount(Dialog, {
      props: {
        title: 'title',
        content: 'content'
      }
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  test('should render prop showCancel=false correctly', () => {
    const wrapper = mount(Dialog, {
      props: {
        title: 'title',
        content: 'content',
        showCancel: false
      }
    })

    expect(wrapper.html()).toMatchSnapshot()
  })
})
