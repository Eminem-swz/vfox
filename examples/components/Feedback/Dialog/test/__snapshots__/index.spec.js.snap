// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dialog should render default correctly 1`] = `
<!--teleport start-->
<!--teleport end-->
`;

exports[`Dialog should render prop showCancel=false correctly 1`] = `
<!--teleport start-->
<!--teleport end-->
`;
